android/library/src/main/java/com/polidea/multiplatformbleadapter/BleAdapter.java

```diff
@@ -2,7 +2,9 @@
 
 interface BleAdapter {
 
-    void createClient(String restoreIdentifierKey);
+    void createClient(String restoreStateIdentifier,
+                      OnEventCallback<String> onAdapterStateChangeCallback,
```

Comment:
Callback is fine. However I would join two callbacks together and call it a `Result`. It's more common name in other languages.

Reply:
We decided to leave it "as it is". Introducing a common "Callback" interface will lead to situation when IntelliJ wraps all anonymous classes provided as parameters to adapter methods and it will show only their methods names as params (and all will be the same), so it can be a little bit confusing at the first moment when you read any adapter method invocation.